pagespeed/kernel/http/response_headers_test.cc

```diff
@@ -2104,4 +2113,32 @@ TEST_F(ResponseHeadersTest, ClearOptionCookies) {
   EXPECT_EQ(kBaseHeaders, headers.ToString());
 }
 
+TEST_F(ResponseHeadersTest, TestHopByHopSanitization) {
+  // RFC hop-by-hop list: http://tools.ietf.org/html/rfc7230#section-6.1
+  ResponseHeaders headers;
+
+  headers.Add(HttpAttributes::kConnection, "Keep-Alive, Foo,, , bar, Cache-Control");
+  headers.Add(HttpAttributes::kKeepAlive, "foo");
+  headers.Add(HttpAttributes::kProxyAuthenticate, "foo");
+  headers.Add(HttpAttributes::kProxyAuthorization, "foo");
+  headers.Add(HttpAttributes::kTE, "foo");
+  headers.Add(HttpAttributes::kTrailers, "foo");
+  headers.Add(HttpAttributes::kTransferEncoding, "foo");
+  headers.Add(HttpAttributes::kUpgrade, "foo");
+  headers.Add(HttpAttributes::kAlternateProtocol, "foo");
+  headers.Add(HttpAttributes::kCacheControl, "foo");
+  // foo: foo is be referenced in "Connection: Foo", and therefore is marked
+  // as hop-by-hop and as such candidate for sanitization.
+  headers.Add("foo", "foo");
+
+  // Before sanitization, we should have headers.
+  EXPECT_NE("HTTP/1.0 0 (null)\r\n\r\n", headers.ToString());
```

Comment:
@jeffkaufman re :"This seems kind of specific for an  EXPECT_NE ?"
I removed this test, I don't think it added much value. Is that ok?